extensions/amp-list/0.1/amp-list.js

```diff
@@ -387,15 +396,44 @@ export class AmpList extends AMP.BaseElement {
           AmpEvents.DOM_UPDATE, /* detail */ null, {bubbles: true});
       this.container_.dispatchEvent(event);
 
-      // Change height if needed.
-      this.measureElement(() => {
-        const scrollHeight = this.container_./*OK*/scrollHeight;
-        const height = this.element./*OK*/offsetHeight;
-        if (scrollHeight > height) {
-          this.attemptChangeHeight(scrollHeight).catch(() => {});
-        }
-      });
+      if (!this.resizeExperimentOn_) {
+        // Change height if needed.
+        this.measureElement(() => {
+          const scrollHeight = this.container_./*OK*/scrollHeight;
+          const height = this.element./*OK*/offsetHeight;
+          if (scrollHeight > height) {
+            this.attemptChangeHeight(scrollHeight).catch(() => {});
+          }
+        });
+      }
     }, this.container_);
+
+    if (this.resizeExperimentOn_ && !this.isSetToLayoutContainer_) {
+      this.mutateElement(() => {
+        this.changeToLayoutContainer_();
+      });
+      this.isSetToLayoutContainer_ = true;
+    }
+  }
+
+  /**
+   * Converts the amp-list to de facto layout container.
+   * @private
+   */
+  changeToLayoutContainer_() {
+    this.element.setAttribute('layout', 'container');
+    const sizer = this.element.querySelector('i-amphtml-sizer');
```

Comment:
We should restrict this to direct children only to avoid removing a child AMP component's sizer.

Reply:
Changed to use `childElementByTag`